indico/modules/categories/controllers/management.py

```diff
@@ -263,10 +263,29 @@ def _get_defaults(self):
 
 
 class RHCreateCategory(RHManageCategoryBase):
+    def _sort_key(self, category):
+        return (natural_sort_key(category.title), category.id)
+
+    def _get_sort_order(self, categories):
+        sorted_categories = sorted(self.category.children, key=self._sort_key)
+        if categories == sorted_categories:
+            return 'asc'
+        elif categories == list(reversed(sorted_categories)):
```

Comment:
```suggestion
        elif categories == sorted_categories[::-1]:
```

untested but i think that does the same ;)

(nothing wrong with your more explicit solution though)